extensions/applicationinsights-analytics-js/src/JavaScriptSDK/AnalyticsPlugin.ts

```diff
@@ -757,6 +686,196 @@ export class ApplicationInsights extends BaseTelemetryPlugin implements IAppInsi
+                    let _history = getHistory();
+
+                    if (isFunction(_history.pushState) && isFunction(_history.replaceState) && typeof Event !== strUndefined) {
+                        _addHistoryListener(extConfig, win, _history, locn);
+                    }
+                }
+            }
+
+            /**
+             * Create a custom "locationchange" event which is triggered each time the history object is changed
+             */
+            function _addHistoryListener(extConfig: IConfig, win: Window, history: History, locn: Location) {
+
+                function _popstateHandler() {
+                    if (_enableAutoRouteTracking) {
+                        _dispatchEvent(win, createDomEvent(extConfig.namePrefix + "locationchange"));
+                    }
+                }
+
+                function _locationChangeHandler() {
+                    // We always track the changes (if the handler is installed) to handle the feature being disabled between location changes
+                    if (_currUri) {
+                        _prevUri = _currUri;
+                        _currUri = locn && locn.href || "";
+                    } else {
+                        _currUri = locn && locn.href || "";
+                    }
+
+                    if (_enableAutoRouteTracking) {
+                        let properties = _self.core.getPlugin<PropertiesPlugin>(PropertiesPluginIdentifier);
+
+                        if (properties) {
+                            let context = properties.plugin.context;
+                            if (context && context.telemetryTrace) {
+                                context.telemetryTrace.traceID = generateW3CId();
+                                let traceLocationName = "_unknown_";
+                                if (locn && locn.pathname) {
+                                    traceLocationName = locn.pathname + (locn.hash || "");
+                                }
+
+                                // This populates the ai.operation.name which has a maximum size of 1024 so we need to sanitize it
+                                context.telemetryTrace.name = dataSanitizeString(_self.diagLog(), traceLocationName);
+                            }
+                        }
+
+                        setTimeout(((uri: string) => {
+                            // todo: override start time so that it is not affected by autoRoutePVDelay
+                            _self.trackPageView({ refUri: uri, properties: { duration: 0 } }); // SPA route change loading durations are undefined, so send 0
+                        }).bind(this, _prevUri), _self.autoRoutePVDelay);
+                    }
+                }
+
+                if (!_historyListenerAdded) {
+                    _addHook(InstrumentFunc(history, "pushState", {
+                        ns: _evtNamespace,
+                        rsp: () => {
+                            if (_enableAutoRouteTracking) {
```

Comment:
This change is specifically to support configuration changes where it might have been initialized with the event handling enabled but disabled later. Rather than removing the hooks, this just stops the hook code form running.